### PR TITLE
fix(annotations): Fix region toggle button does not exit region mode

### PR DESCRIPTION
**Pull request by mxiao6**

### CODE REVIEW

File: src/lib/viewers/BaseViewer.js [hunk 1]
```diff
@@ -1055,10 +1055,13 @@ class BaseViewer extends EventEmitter {
...
      *
      * @private
+     * @param {boolean} activeControl - avtive control mode in AnnotationControls
      * @return {void}
      */
-    handleRegionClick() {
-        this.annotator.toggleAnnotationMode(AnnotationMode.REGION);
+    handleRegionClick({ activeControl }) {
+        if (activeControl === AnnotationMode.REGION) {
+            this.annotator.toggleAnnotationMode(AnnotationMode.REGION);
```
Comment (jstoffan) [inline src/lib/viewers/BaseViewer.js:1063]: Would we need to explicitly set it back to AnnotationMode.NONE?

Comment (mxiao6) [reply src/lib/viewers/BaseViewer.js:1063]: It's already set to `AnnotationMode.NONE` in `resetControls`.

File: src/lib/viewers/BaseViewer.js [hunk 2]
```diff
@@ -1055,10 +1055,13 @@ class BaseViewer extends EventEmitter {
      * Handler for annotation toolbar region comment button click event.
      *
      * @private
+     * @param {boolean} activeControl - avtive control mode in AnnotationControls
```
Comment (ConradJChan) [inline src/lib/viewers/BaseViewer.js:1058]: `acvtive` -> `active`.

I believe we typically represent destructured arguments as:
`@param {boolean} options.activeControl - ...`

Comment (mxiao6) [reply src/lib/viewers/BaseViewer.js]: Thank you for your feedback, I'll pay attention to those in my future PRs. I just removed this line in my latest push.

File: src/lib/AnnotationControls.ts [hunk 3]
```diff
@@ -176,7 +175,7 @@ export default class AnnotationControls {
     /**
      * Initialize the annotation controls with options.
      */
-    public init({ onRegionClick = noop, onReset = noop }: Options = {}): void {
+    public init({ onRegionClick = noop, onEscape = noop }: Options = {}): void {
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:178]: Can we alphabetize these parameters?

File: src/lib/AnnotationControls.ts [hunk 4]
```diff
@@ -146,28 +144,29 @@ export default class AnnotationControls {
...
         this.resetControls();
 
         if (prevActiveControl !== mode) {
-            this.currentActiveControl = mode as AnnotationMode;
+            this.currentMode = mode as AnnotationMode;
             this.controlsMap[mode]();
         }
 
-        onClick({ activeControl: this.currentActiveControl, event });
+        onClick({ activeControl: this.currentMode, event });
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:156]: Is `activeControl` used anywhere? If not, can it be removed? If so, can it be renamed?

File: src/lib/AnnotationControls.ts [hunk 5]
```diff
@@ -146,28 +144,29 @@ export default class AnnotationControls {
      * Region comment button click handler
      */
     private handleClick = (onClick: ClickHandler, mode: AnnotationMode) => (event: MouseEvent): void => {
-        const prevActiveControl = this.currentActiveControl;
+        const prevActiveControl = this.currentMode;
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:147]: `previousMode`?

File: src/lib/AnnotationControls.ts [hunk 6]
```diff
@@ -106,23 +106,21 @@ export default class AnnotationControls {
     private handleFullscreenExit = (): void => this.handleFullscreenChange(false);
 
     public getActiveMode = (): AnnotationMode => {
```
Comment (jstoffan) [inline src/lib/AnnotationControls.ts:108]: Is this used anywhere?

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 7]
```diff
@@ -1100,7 +1100,7 @@ class DocBaseViewer extends BaseViewer {
         if (this.areNewAnnotationsEnabled() && this.hasAnnotationCreatePermission()) {
             this.annotationControls.init({
                 onRegionClick: this.handleRegionClick,
-                onReset: this.handleAnnotationControlsReset,
+                onEscape: this.handleAnnotationControlsEscape,
```
Comment (jstoffan) [inline src/lib/viewers/doc/DocBaseViewer.js:1103]: Alphabetize in all files.

**Pull request merged**